### PR TITLE
Barrel Services for the MFT

### PR DESCRIPTION
**Pull request by rhelacon**
Implementation of the class Barrel, that describes the BarrelServices for the MFT, overlaps fixed.

### CODE REVIEW

**Comment by rhelacon** [comment]
Can I have this reviewed or merged? the error looks like it's not related with the code, the final position will be added in GeometryBuilder.cxx later @shahor02

File: Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include "TNamed.h"
+
+class TGeoVolumeAssembly;
+
+namespace o2
+{
+namespace mft
+{
+
+class Barrel : public TNamed
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h]: prefer not to inherit from TNamed classes if possible.

File: Detectors/ITSMFT/MFT/base/src/Barrel.cxx [hunk 2]
```diff
@@ -0,0 +1,505 @@
...
+using namespace o2::mft;
+
+ClassImp(o2::mft::Barrel);
+
+//_____________________________________________________________________________
+Barrel::Barrel() : TNamed(), mBarrel(nullptr)
+{
+
+  // default constructor
+}
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/base/src/Barrel.cxx]: could be done in header file `Barrel() = default;`

File: Detectors/ITSMFT/MFT/base/src/Barrel.cxx [hunk 3]
```diff
@@ -0,0 +1,505 @@
...
+
+//_____________________________________________________________________________
+Barrel::~Barrel() = default;
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/base/src/Barrel.cxx]: could be done in header

File: Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h [hunk 4]
```diff
@@ -25,7 +25,7 @@ class Barrel : public TNamed
 
  public:
   Barrel() = default;
-  ~Barrel() = default;
+  ~Barrel() = override;
```
Comment (sawenzel) [inline Detectors/ITSMFT/MFT/base/include/MFTBase/Barrel.h]: override is only needed for virtual classes. Do you need Barrel to be virtual (aka do you need to inherit from TNamed?).

**Comment by rhelacon** [comment]
@sawenzel  I got a warning and an error when I tried   ~Barrel() = default; in the o2checkcode, I don't understand why exactly o2checkcode prefers using override instead to define with default (up to the point to put an x and mark as an error), so I had to leave as virtual and then define it in the .cxx file in order to get the check.
 About the inheritance most of our classes are inherited from TNamed, I guess this help when we are rendering the geometry and place them, but I'm not 100% sure, so that's why I would like the opinion of @frmanso since he has plenty experience in the implementation of these classes, if he says it's not needed them I'll follow your suggestion

**Comment by aphecetche** [comment]
@rhelacon your PR has now a conflict due to the fact that the cmake migration branch has been merged. CMakeLists.txt are now different, see https://github.com/AliceO2Group/AliceO2/blob/dev/doc/CMakeInstructions.md. I can help you (today or tomorrow) with the transition if need be.

**Comment by rhelacon** [comment]
Many thanks @aphecetche... I think will be easier to make a new branch and a new pull request from there because I got behind for many commits from the dev branch and merging it's not quite easy when a change has a huge impact

**Pull request closed without merging**